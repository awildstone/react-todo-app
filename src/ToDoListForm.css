.new-todo {
	margin: 20px;
}
