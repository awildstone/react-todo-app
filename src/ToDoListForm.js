import React, { useState } from 'react';
import { nanoid } from 'nanoid';
import './ToDoListForm.css'

const ToDoListForm = ({ addToDo }) => {
    const INITIAL_STATE = { toDo: ''};

    const [formData, setFormData] = useState('');

    const handleChange = (evt) => {
        const { name, value } = evt.target;
        setFormData(formData => ({...formData, [name]: value}));
    }
    
    const handleSubmit = (evt) => {
        evt.preventDefault();
        addToDo({ id: nanoid(), ...formData });
        evt.target.reset();
        setFormData(INITIAL_STATE);
    }

    return (
        <div className="new-todo">
            <form onSubmit={handleSubmit}>
                <label htmlFor="task">Task: </label>
                <input
                    id="task"
                    type="text"
                    name="task"
                    onChange={handleChange}
                />
                <button className="save">Add</button>
            </form>
        </div>
    )
}

export default ToDoListForm;

