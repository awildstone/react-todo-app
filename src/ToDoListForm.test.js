import React from 'react';
import { render } from '@testing-library/react';
import ToDoListForm from './ToDoListForm';

it('renders a ToDoListForm without crashing', () => {
    render(<ToDoListForm />);
});

it('matches a ToDoListForm snapshot', () => {
    const { asFragment } = render(<ToDoListForm />);
    expect(asFragment()).toMatchSnapshot();
});