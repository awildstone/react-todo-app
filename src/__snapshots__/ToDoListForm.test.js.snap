// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a ToDoListForm snapshot 1`] = `
<DocumentFragment>
  <div
    class="new-todo"
  >
    <form>
      <label
        for="task"
      >
        Task: 
      </label>
      <input
        id="task"
        name="task"
        type="text"
      />
      <button
        class="save"
      >
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
